### PR TITLE
feat: develop pose with obstacle command generator

### PR DESCRIPTION
**Pull request by yrh012**
## Description

- #48 

This PR addresses the issue by introducing a new command generator class that randomly spawns obstacles within a specified range near each generated command. It is parameterized to allow users to define the probability of spawning an obstacle in a given environment, the range within which the obstacle is placed around the target, and to restrict the commands to position setpoints only, excluding orientation setpoints. It also includes an exclusion region so that the users can avoid spawning obstacles that are already in collision with the robot.

## Review guidelines

Estimated Time of Review: 10 minutes

## Checklist before merging:

- [ ] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

File: source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py [hunk 1]
```diff
@@ -129,6 +130,91 @@ class Ranges:
...
+    goal_pose_visualizer_cfg: VisualizationMarkersCfg = FRAME_MARKER_CFG.replace(prim_path="/Visuals/Command/goal_pose")
+    """The configuration for the goal pose visualization marker. Defaults to FRAME_MARKER_CFG."""
+
+    current_pose_visualizer_cfg: VisualizationMarkersCfg = FRAME_MARKER_CFG.replace(
+        prim_path="/Visuals/Command/body_pose"
+    )
+    """The configuration for the current pose visualization marker. Defaults to FRAME_MARKER_CFG."""
+
+    # Set the scale of the visualization markers to (0.1, 0.1, 0.1)
+    goal_pose_visualizer_cfg.markers["frame"].scale = (0.1, 0.1, 0.1)
```
[the remark this answers is not in the document]
Comment (yrh012) [reply source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py:214]: yes the scale depends on the asset and this is a custom visualization asset for the 6D pose (i.e. has the 3 colored axis, red, green, blue)

File: source/isaaclab/isaaclab/envs/mdp/commands/pose_obstacle_command.py [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+            random_range[mask_right] = (
+                excluding_high + ((u_right - p_left) / (1.0 - p_left)) * right_len
+            )
+
+        return random_range
+
+    @staticmethod
+    def _randomly_remove_obstacles(
+        positions: torch.Tensor,
+        probability_of_obstacle_existance: float = 0.5,
```
[the remark this answers is not in the document]
Comment (yrh012) [reply source/isaaclab/isaaclab/envs/mdp/commands/pose_obstacle_command.py:180]: Yep makes sense :+1:

File: source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yrh012) [reply source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py:214]: image
Regarding color, it is a combination of three different assets and the alpha is given in a sort of color diffusion scale and I think since those are like a "standard" axes colors, we keep the original pure color, what do you think?

[the remark this answers is not in the document]
Comment (bpapaspyros) [reply source/isaaclab/isaaclab/envs/mdp/commands/commands_cfg.py:214]: that's fine honestly, mostly asking out of curiosity, no action needed here honestly, just trying to put the scales together

File: source/isaaclab/isaaclab/envs/mdp/commands/pose_obstacle_command.py [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+        if mask_right.any():
+            u_right = u[mask_right]
+            random_range[mask_right] = (
+                excluding_high + ((u_right - p_left) / (1.0 - p_left)) * right_len
+            )
+
+        return random_range
+
+    @staticmethod
+    def _randomly_remove_obstacles(
```
[the remark this answers is not in the document]
Comment (bpapaspyros) [reply source/isaaclab/isaaclab/envs/mdp/commands/pose_obstacle_command.py:180]: reason I am bringing this up is because the way I picture it, those zones we are (over)excluding are quite big. Ultimately, there are more valid ones, but we are still cutting out a lot of near-ee samples that might realistically be quite important.

[the remark this answers is not in the document]
Comment (yrh012) [reply source/isaaclab/isaaclab/envs/mdp/commands/pose_obstacle_command.py:180]: In theory, I can select randomly an axis and then make sure that this axis is made sure that it satisfied the exclusion criteria instead of the 3 axes, what do you think of that?

[the remark this answers is not in the document]
Comment (bpapaspyros) [reply source/isaaclab/isaaclab/envs/mdp/commands/pose_obstacle_command.py:180]: (the naive approach I mentioned might cost in time if that's done very often though. So if you have a better analytical approach that would make sense, yes)

**Review by bpapaspyros — APPROVED** [review]
looks good, thanks for checking the sampling again. As long as this is tested, good for me (just a small last question).

File: source/isaaclab/isaaclab/envs/mdp/commands/pose_obstacle_command.py [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+                *self.cfg.ranges.obstacle_pos_z,
+            )
+
+            orientations = torch.zeros((len(env_ids), 4), device=rigid_body.device)
+            orientations[:, 0] = 1
+            rigid_body.write_root_pose_to_sim(
+                torch.cat([positions, orientations], dim=-1), env_ids=env_ids
+            )
+
+    def _update_command(self):
```
[the remark this answers is not in the document]
Comment (yrh012) [reply source/isaaclab/isaaclab/envs/mdp/commands/pose_obstacle_command.py:316]: In their base class, this is an abstract method is it is not overriding but rather implementing it.
    @abstractmethod
    def _update_command(self):
        """Update the command based on the current state."""
        raise NotImplementedError

[the remark this answers is not in the document]
Comment (yrh012) [reply source/isaaclab/isaaclab/envs/mdp/commands/pose_obstacle_command.py:316]: hahaha in some sense yes :+1:
